src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -2,16 +2,19 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Buffers;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
+using System.Runtime.CompilerServices;
 
 namespace System.Numerics
 {
     [Serializable]
     [System.Runtime.CompilerServices.TypeForwardedFrom("System.Numerics, Version=4.0.0.0, PublicKeyToken=b77a5c561934e089")]
     public readonly struct BigInteger : IFormattable, IComparable, IComparable<BigInteger>, IEquatable<BigInteger>
     {
+        private const int StackAllocThreshold = 256;
```

Comment:
As said before I think this value is too high. For `uint` it results in `256 * sizeof(uint) = 256 * 4 = 1024`.
```suggestion
        private const int StackAllocThreshold = 256 / sizeof(uint);
```
To have a "de-facto" size of 256 bytes.

Reply:
Same value used by original authors of `BigIntegerCalculator`. Moreover, this constant and static field in `BigIntegerCalculator` should be merged into single one. This requires discussion because it affects unit tests.

Reply:
In `BitInteger` we have potential more stack-allocs than before, so it should be on safe side without potential stackoverflows.

We'll see which value is the proper one...

Reply:
Anyway, we need to decide what to do with two constants with the same meaning. Obviously, this is not a final commit so I'll change it to appropriate value after discussion.

Reply:
And, as I said `BigIntegerCalculator.AllocationThreshold` is not a constant so there is no positive effect on stack allocation operator. I have to wait for decision. IMO, I would change it to constant and remove the code from unit tests + move `BigIntegerCalculator` to `BigInteger` as inner class to share the same constant because I don't want to change visibility scope of the constant from **private** to **internal**.

Reply:
The threshold for `_malloca` on Windows is `#define _ALLOCA_S_THRESHOLD 1024`, so I'm not convinced that `1024` is "too" big. It may be a problem if multiple layers of a function try to allocate that much, but it shouldn't be if you know this will be the only "big" allocation. For example, `double.Parse` allocates a `769` byte buffer today, but the method is essentially a "leaf method" and so the total set of allocations past that is basically well known.

Also, this would ideally be something more centrally managed for the framework ("give me a span on either the stack or heap, based on some limit"), but I think that would be a separate issue/PR.

Reply:
Managed code and native code are also different, so while `1024` might be a good number for `native`, it might be too big for `managed` and the exact threshold might require more though/consideration.

Reply:
> ("give me a span on either the stack or heap, based on some limit"),

https://github.com/dotnet/runtime/issues/25423